transport/socks5/stream_dialer.go

```diff
@@ -23,26 +23,59 @@ import (
 	"github.com/Jigsaw-Code/outline-sdk/transport"
 )
 
+// https://datatracker.ietf.org/doc/html/rfc1929
+// Credentials can be nil, and that means no authentication.
+type Credentials struct {
+	username []byte
+	password []byte
+}
+
+func NewCredentials(username, password string) (Credentials, error) {
+	var c Credentials
+	usernameBytes := []byte(username)
+	passwordBytes := []byte(password)
+
+	if len(usernameBytes) > 255 {
+		return c, errors.New("username exceeds 255 bytes")
+	}
+	if len(usernameBytes) == 0 {
```

Comment:
Let's remove this check. I don't think this is an RFC requirement. Same for the password.